org.osgi.test.common/src/main/java/org/osgi/test/common/osgi/ServiceUse.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) OSGi Alliance (2019). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.osgi;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Filter;
+import org.osgi.framework.ServiceReference;
+import org.osgi.util.tracker.ServiceTracker;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+
+public class ServiceUse<T> implements AutoCloseable {
+
+	public static final long		DEFAULT_TIMEOUT	= 200;
+
+	private final int				cardinality;
+	private final Filter			filter;
+	private final long				timeout;
+
+	private ServiceTracker<T, T>	tracker;
+
+	public ServiceUse(Filter filter, int cardinality, long timeout) {
+		this.filter = requireNonNull(filter);
+
+		if (cardinality < 0) {
+			throw new IllegalArgumentException("cardinality must be zero or greater");
+		}
+		this.cardinality = cardinality;
+
+		if (timeout < 0) {
+			throw new IllegalArgumentException("timeout must be zero or greater");
+		}
+		this.timeout = timeout;
+	}
+
+	@Override
+	public void close() throws Exception {
+		tracker.close();
```

Comment:
You can NPE here since tracker is null until init is called.

Reply:
fixing..